packages/taquito/src/operations/register-global-constant-operation.ts

```diff
@@ -1,71 +1,93 @@
-    constructor(
-        hash: string,
-        private readonly params: RPCRegisterGlobalConstantOperation,
-        public readonly source: string,
-        raw: ForgedBytes,
-        results: OperationContentsAndResult[],
-        context: Context
-    ) {
-        super(hash, raw, results, context);
+    this.globalConstantHash = this.operationResults && this.operationResults.global_address;
+  }
 
-        this.globalConstantHash = this.operationResults && this.operationResults.global_address;
-    }
+  get operationResults() {
+    const registerGlobalConstantOp =
+      Array.isArray(this.results) &&
+      (this.results.find(
+        (op) => op.kind === 'register_global_constant'
+      ) as OperationContentsAndResultRegisterGlobalConstant);
+    const result =
+      registerGlobalConstantOp &&
+      registerGlobalConstantOp.metadata &&
+      registerGlobalConstantOp.metadata.operation_result;
+    return result ? result : undefined;
+  }
 
-    get operationResults() {
-        const registerGlobalConstantOp =
-            Array.isArray(this.results) &&
-            (this.results.find(op => op.kind === 'register_global_constant') as OperationContentsAndResultRegisterGlobalConstant);
-        const result = registerGlobalConstantOp && registerGlobalConstantOp.metadata && registerGlobalConstantOp.metadata.operation_result;
-        return result ? result : undefined;
+  get status() {
+    const operationResults = this.operationResults;
+    if (operationResults) {
+      return operationResults.status;
+    } else {
+      return 'unknown';
     }
+  }
 
-    get status() {
-        const operationResults = this.operationResults;
-        if (operationResults) {
-            return operationResults.status;
-        } else {
-            return 'unknown';
-        }
-    }
+  get registeredExpression() {
+    return this.params.value;
+  }
 
-    get registeredExpression() {
-        return this.params.value;
-    }
+  get fee() {
+    return this.params.fee;
+  }
 
-    get fee() {
-        return this.params.fee;
-    }
+  get gasLimit() {
+    return this.params.gas_limit;
+  }
 
-    get gasLimit() {
-        return this.params.gas_limit;
-    }
+  get storageLimit() {
+    return this.params.storage_limit;
+  }
 
-    get storageLimit() {
-        return this.params.storage_limit;
-    }
+  get errors() {
+    return this.operationResults && this.operationResults.errors;
+  }
 
-    get errors() {
-        return this.operationResults && this.operationResults.errors;
-    }
+  get consumedGas() {
+    BigNumber.config({ DECIMAL_PLACES: 0, ROUNDING_MODE: BigNumber.ROUND_UP });
+    return this.consumedMilliGas
+      ? new BigNumber(this.consumedMilliGas).dividedBy(1000).toString()
+      : undefined;
+  }
+
+  get consumedMilliGas() {
+    const consumedMilliGas = this.operationResults && this.operationResults.consumed_milligas;
+    return consumedMilliGas ? consumedMilliGas : undefined;
```

Comment:
`return this.operationResults?.consumed_milligas`

Reply:
yep better like this, thanks